### PR TITLE
Issue #311 playback input improvements

### PR DESCRIPTION
**Pull request by akinwale**
- Pressing the space bar can initiate video playback
- Pressing the space bar can pause and resume video playback
- Clicking on the video area (above the video controls) after initial playback can pause and resume video playback

### CODE REVIEW

File: ui/js/component/video/internal/play-button.jsx [hunk 1]
```diff
@@ -4,12 +4,23 @@ import Link from "component/link";
 import Modal from "component/modal";
 
 class VideoPlayButton extends React.PureComponent {
+  componentDidMount() {
+    document.addEventListener("keydown", this.onKeyDown.bind(this));
```
Comment (kauffj) [inline ui/js/component/video/internal/play-button.jsx]: this needs to be undone in unmount

File: ui/js/component/video/internal/play-button.jsx [hunk 2]
```diff
@@ -4,12 +4,23 @@ import Link from "component/link";
...
+  }
+
   onPurchaseConfirmed() {
     this.props.closeModal();
     this.props.startPlaying();
     this.props.loadVideo(this.props.uri);
   }
 
+  onKeyDown(event) {
+    if (event.keyCode === 32) {
```
Comment (kauffj) [inline ui/js/component/video/internal/play-button.jsx]: Can we use a code or constant here? Possible https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code?

File: ui/js/component/video/internal/player.jsx [hunk 3]
```diff
@@ -29,10 +29,11 @@ class VideoPlayer extends React.PureComponent {
     player.append(
       this.file(),
       container,
-      { autoplay: false, controls: true },
+      { autoplay: false, controls: true, enableKeyboard: true },
       renderMediaCallback.bind(this)
     );
 
+    document.addEventListener("keydown", this.togglePlay.bind(this));
```
Comment (kauffj) [inline ui/js/component/video/internal/player.jsx]: This needs to be unbound when unmounted.

**Comment by akinwale** [comment]
@kauffj Pushed a new commit with the requested changes.

**Comment by kauffj** [comment]
@akinwale changes look good, thanks! One issue I noticed while testing is that the "keydown" bind hijacks the ability to type space into the omnibar.

Either the event needs to be bound to an element deeper in the DOM, or we need to check if the keypress is happening inside of a form element and allow it to proceed rather than treating is a possible play press.

**Comment by akinwale** [comment]
@kauffj Added a form control check in the new commit.

**Pull request merged**